java-manta-client/src/main/java/com/joyent/manta/http/EncryptionHttpHelper.java

```diff
@@ -866,4 +989,34 @@ private Cipher buildMetadataDecryptCipher(final SupportedCipherDetails cipherDet
         }
         return metadataCipher;
     }
+
+    /**
+     * Converts a nullable {@link Long} array into a long primitive array.
+     * Unlimited positions are represented as the hard file size limits of the
+     * specified cipher.
+     *
+     * @param ranges array containing two elements
+     * @param cipherDetails cipher being used to compute range
+     * @return updated range coordinates based on cipher/cipher mode configuration
+     */
+    private static long[] byteRangeAsNullSafe(final Long[] ranges,
```

Comment:
I changed this to package scope. @geek can you add some additional testing to verify this function's behavior?